core/domain/suggestion_registry.py

```diff
@@ -1095,3 +1095,107 @@ def reject(self, reviewer_id, rejection_message):
+            suggestions that are currently in review.
+    """
+
+    def __init__(
+            self, translation_reviewer_counts_by_lang_code,
+            translation_suggestion_counts_by_lang_code,
+            question_reviewer_count, question_suggestion_count):
+        self.translation_reviewer_counts_by_lang_code = (
+            translation_reviewer_counts_by_lang_code
+        )
+        self.translation_suggestion_counts_by_lang_code = (
+            translation_suggestion_counts_by_lang_code
+        )
+        self.question_reviewer_count = question_reviewer_count
+        self.question_suggestion_count = question_suggestion_count
+
+    def validate(self):
+        """Validates the ReviewerAndSuggestionCounts object.
+
+        Raises:
+            ValidationError. One or more attributes of the
+                ReviewerAndSuggestionCounts object is invalid.
+        """
+        for language_code, reviewer_count in (
+                self.translation_reviewer_counts_by_lang_code.items()):
+            if reviewer_count < 0:
+                raise utils.ValidationError(
+                    'Expected the translation reviewer count to be '
+                    'non-negative, recieved: %s. The language code for the '
+                    'translation was %s.' % (reviewer_count, language_code)
+                )
+            if not utils.is_valid_language_code(language_code):
+                raise utils.ValidationError(
+                    'Invalid language code for the translation reviewer '
+                    'counts: %s.' % language_code)
+
+        for language_code, suggestion_count in (
+                self.translation_suggestion_counts_by_lang_code.items()):
+            if suggestion_count < 0:
+                raise utils.ValidationError(
+                    'Expected the translation suggestion count to be '
+                    'non-negative, recieved: %s. The language code for the '
+                    'translation was %s.' % (suggestion_count, language_code)
+                )
+            if not utils.is_valid_language_code(language_code):
+                raise utils.ValidationError(
+                    'Invalid language code for the translation suggestion '
+                    'counts: %s.' % language_code)
+
+        if self.question_reviewer_count < 0:
+            raise utils.ValidationError(
+                'Expected the question reviewer count to be non-negative, '
+                'recieved: %s.' % (self.question_reviewer_count)
+            )
+
+        if self.question_suggestion_count < 0:
+            raise utils.ValidationError(
+                'Expected the question suggestion count to be non-negative, '
+                'recieved: %s.' % (self.question_suggestion_count)
+            )
+
+    def set_translation_reviewer_count_for_language_code(
+            self, language_code, count):
+        """Sets the translation reviewer count to be count, for the language
+        code given.
```

Comment:
for the given language count*

Reply:
This is correct the way it is right? The language code is given.

Reply:
I'm fine with either. (Generally, we have `for the given ....` in the codebase!) :)

Reply:
I'll change this in the follow up PR :)